pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -388,19 +397,65 @@ func (r *Reconciler) streamLogs(ctx context.Context, o results.Object, logType,
 		return fmt.Errorf("error reading from tkn reader: %w", err)
 	}
 
-	errChanRepeater := make(chan error)
-	go func(echan <-chan error, o metav1.Object) {
-		writeErr := <-echan
-		errChanRepeater <- writeErr
+	errChanRepeater := make(chan error, 100)
 
-		_, err := writer.Flush()
+	// logctx is derived from ctx. Therefore, if ctx is cancelled (either explicitly through a call to its cancel
+	// function or when it reaches its deadline), logctx will be cancelled automatically.
+	logctx, _ := context.WithTimeout(ctx, 10*time.Minute)
+
+	go func(ctx context.Context, echan <-chan error, o metav1.Object) {
+		defer close(errChanRepeater)
+		select {
+		case <-ctx.Done():
+			logger.Warnw("Context done streaming log",
+				zap.String("namespace", o.GetNamespace()),
+				zap.String("name", o.GetName()),
+			)
+		case writeErr := <-echan:
+			errChanRepeater <- writeErr
+			var gofuncerr error
+			_, gofuncerr = writer.Flush()
+			if gofuncerr != nil {
+				logger.Error(gofuncerr)
+			}
+			if gofuncerr = logsClient.CloseSend(); gofuncerr != nil {
+				logger.Error(gofuncerr)
+			}
+			logger.Debugw("Flush in streamLogs done",
+				zap.String("namespace", o.GetNamespace()),
+				zap.String("name", o.GetName()),
+			)
+		}
+		logger.Debugw("Gofunc in streamLogs done",
+			zap.String("namespace", o.GetNamespace()),
+			zap.String("name", o.GetName()),
+		)
+
+		err = r.RemoveFinalizer(ctx, o)
 		if err != nil {
-			logger.Error(err)
+			logger.Errorw("Error removing finalizer",
+				zap.String("namespace", o.GetNamespace()),
+				zap.String("name", o.GetName()),
+				zap.Error(err),
+			)
+			return
 		}
-		if err = logsClient.CloseSend(); err != nil {
-			logger.Error(err)
+
+		logger.Debugw("Finalizer removed successfully",
```

Comment:
while @pmacik is testing for us, let's temporarily change this to an Info level log

that way, if need be, we can look for these message in the results watcher log gathered in the artifacts from his e2e tests